### PR TITLE
Updates on Cryptographic Storage Cheat Sheet

### PR DESCRIPTION
**Pull request by guerilla7**
- [x] All the markdown files do not raise any validation policy violation, see policy [here](https://github.com/OWASP/CheatSheetSeries#editor--validation-policy).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries#conversion-rules).
- [x] Any references to website have been formatted as [TEXT](URL)
- [x] You verified/tested the effectiveness of your contribution (e.g.: defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://travis-ci.org/OWASP/CheatSheetSeries/pull_requests).

This PR covers issue #181

### CODE REVIEW

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 1]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
 
 **Note:** Do not use [ECB mode](http://en.wikipedia.org/wiki/Block_cipher_modes_of_operation#Electronic_codebook_.28ECB.29) for encrypting lots of data (the other modes are better because they chain the blocks of data together to improve the data security).
 
-#### Rule - Use strong random numbers
+#### Rule - Use cryptographically secure pseudo random number generators (CSPRNG)
 
-Ensure that all random numbers, especially those used for cryptographic parameters (keys, IV’s, MAC tags), random file names, random GUIDs, and random strings are generated in a cryptographically strong fashion.
+When generating random bits in the context of sofware security, use the available Cryptographically Secure Pseudo- Random Number Generator (CSPRNG) library or module
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Remove the space between *Pseudo* and *Random*.

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 2]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+available on the programming language you are using.  
+
+Do not use Pseudo-Random Number Generators (PRNG) since they are not secure and will produce deterministic results. The chosen numbers are not completely random because a mathematical algorithm is used to select them, but they are sufficiently random for practical purposes. 
+
+PRNGs are primarily used for generating random-looking numbers in use cases such as modeling and simulation in software,while CSPRNGs are intended for generating random-looking numbers resistant to prediction for security use cases such as random passwords, CSRF tokens, session IDs, etc.
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a space before *while CSPRNGs are...*

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 3]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
 
-The following functions are considered **weak** random number generators and should not be used.
+The following functions are considered **weak** pseudo-random number generators and should not be used in the context of security:
 
 - C : `random()`, `rand()` instead use [getrandom(2)](http://man7.org/linux/man-pages/man2/getrandom.2.html)
-- Java : `java.util.Random()` instead use `java.security.SecureRandom`
+- Java and Android OS : `java.util.Random()` instead use `java.security.SecureRandom`
 - PHP : `rand()`, `mt_rand()`, `array_rand()`, `uniqid()` instead use [random_bytes()](https://www.php.net/manual/en/function.random-bytes.php), [random_int()](https://www.php.net/manual/en/function.random-int.php) in PHP 7 or [openssl_random_pseudo_bytes()](https://www.php.net/manual/en/function.openssl-random-pseudo-bytes.php) in PHP 5 (which is **deprecated** and **should not be used**)
+- .NET/C# : `Random()`, instead use `RNGCryptoServiceProvider`
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a link to the official documentation of the recommended class (like it is done for technologies above).

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 4]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+-  Objective-C/Apple IOS : `arc4random()` (Uses RC4 Cipher), instead use `SecRandomCopyBytes`
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a link to the official documentation of the recommended class (like it is done for technologies above).

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 5]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+- Python : `random()`, instead use `secrets()`
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a link to the official documentation of the recommended class (like it is done for technologies above).

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 6]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+- Ruby : `Random`, instead use `SecureRandom`
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a link to the official documentation of the recommended class (like it is done for technologies above).

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 7]
```diff
@@ -55,19 +55,31 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+- Go: `rand` using `math/rand` package, instead use `crypto.rand` package
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Add a link to the official documentation of the recommended class (like it is done for technologies above).

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 8]
```diff
@@ -55,19 +55,32 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
 Ensure that random algorithms are seeded with sufficient entropy.
 
 Tools like [NIST RNG Test tool](http://csrc.nist.gov/groups/ST/toolkit/rng/documentation_software.html) (as used in PCI PTS Derived Test Requirements) can be used to comprehensively assess the quality of a Random Number Generator by reading e.g. 128MB of data from the RNG source and then assessing its randomness properties with the tool.
 
-The following functions are considered **weak** random number generators and should not be used.
+The following functions are considered **weak** pseudo-random number generators and should not be used in the context of security:
 
 - C : `random()`, `rand()` instead use [getrandom(2)](http://man7.org/linux/man-pages/man2/getrandom.2.html)
-- Java : `java.util.Random()` instead use `java.security.SecureRandom`
+- Java and Android OS : `java.util.Random()` instead use `java.security.SecureRandom`
```
Comment (righettod) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Sorry, i have missed it, can you add the link to the javadoc for `java.security.SecureRandom`?

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 9]
```diff
@@ -55,19 +55,32 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
 
 **Note:** Do not use [ECB mode](http://en.wikipedia.org/wiki/Block_cipher_modes_of_operation#Electronic_codebook_.28ECB.29) for encrypting lots of data (the other modes are better because they chain the blocks of data together to improve the data security).
 
-#### Rule - Use strong random numbers
+#### Rule - Use cryptographically secure pseudo random number generators (CSPRNG)
 
-Ensure that all random numbers, especially those used for cryptographic parameters (keys, IV’s, MAC tags), random file names, random GUIDs, and random strings are generated in a cryptographically strong fashion.
+When generating random bits in the context of sofware security, use the available Cryptographically Secure Pseudo-Random Number Generator (CSPRNG) library or module
+available on the programming language you are using.
```
Comment (ThunderSon) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: Remove the new line and set them on the same line

File: cheatsheets/Cryptographic_Storage_Cheat_Sheet.md [hunk 10]
```diff
@@ -55,19 +55,32 @@ In general, you should not use AES, DES or other symmetric cipher primitives dir
...
+When generating random bits in the context of sofware security, use the available Cryptographically Secure Pseudo-Random Number Generator (CSPRNG) library or module
+available on the programming language you are using.  
+
+Do not use Pseudo-Random Number Generators (PRNG) since they are not secure and will produce deterministic results. The chosen numbers are not completely random because a mathematical algorithm is used to select them, but they are sufficiently random for practical purposes. 
+
+PRNGs are primarily used for generating random-looking numbers in use cases such as modeling and simulation in software, while CSPRNGs are intended for generating random-looking numbers resistant to prediction for security use cases such as random passwords, CSRF tokens, session IDs, etc.
+
+The use of CSPRNGs will ensure that all random numbers, especially those used for cryptographic parameters (keys, IV’s, MAC tags), random file names, random GUIDs, and random strings are generated in a cryptographically strong fashion.
```
Comment (ThunderSon) [inline cheatsheets/Cryptographic_Storage_Cheat_Sheet.md]: IVs and not IV's

**Comment by ThunderSon** [comment]
closes #181

**Pull request merged**